web_src/js/utils.test.js

```diff
@@ -138,12 +138,12 @@ test('encodeURLEncodedBase64, decodeURLEncodedBase64', () => {
   expect(encodeURLEncodedBase64(uint8array('AA?'))).toEqual('QUE_'); // standard base64: "QUE/"
   expect(encodeURLEncodedBase64(uint8array('AA~'))).toEqual('QUF-'); // standard base64: "QUF+"
 
-  expect(decodeURLEncodedBase64('QUE/')).toEqual(uint8array('AA?'));
-  expect(decodeURLEncodedBase64('QUF+')).toEqual(uint8array('AA~'));
-  expect(decodeURLEncodedBase64('QUE_')).toEqual(uint8array('AA?'));
-  expect(decodeURLEncodedBase64('QUF-')).toEqual(uint8array('AA~'));
+  expect(decodeURLEncodedBase64('QUE/')).deep.equal(uint8array('AA?'));
+  expect(decodeURLEncodedBase64('QUF+')).deep.equal(uint8array('AA~'));
+  expect(decodeURLEncodedBase64('QUE_')).deep.equal(uint8array('AA?'));
+  expect(decodeURLEncodedBase64('QUF-')).deep.equal(uint8array('AA~'));
 
   expect(encodeURLEncodedBase64(uint8array('a'))).toEqual('YQ'); // standard base64: "YQ=="
-  expect(decodeURLEncodedBase64('YQ')).toEqual(uint8array('a'));
-  expect(decodeURLEncodedBase64('YQ==')).toEqual(uint8array('a'));
+  expect(decodeURLEncodedBase64('YQ')).deep.equal(uint8array('a'));
+  expect(decodeURLEncodedBase64('YQ==')).deep.equal(uint8array('a'));
```

Comment:
Why these changes? As I understand `toEqual` is already performing deep equality checks, unless to checked data is an error, so I prefer to keep using it for everything. From [Reference](https://vitest.dev/api/expect.html#toequal):

> A deep equality will not be performed for Error objects. To test if something was thrown, use [toThrowError](https://vitest.dev/api/expect.html#tothrowerror) assertion.

Reply:
My guess: Because we are using an `array` here, and not a string like above.
Hence, only the array pointer would be compared and not the array contents.

Reply:
> Hence, only the array pointer would be compared and not the array contents.

Really? Difference arrays usually mean different "pointer", how did the tests pass?

Reply:
:shrug: as I said, that's only my guess why these changes were necessary.
@puni9869 will know why they were needed. Perhaps jest made a breaking change in this minor update?

Reply:
> > Hence, only the array pointer would be compared and not the array contents.
> 
> Really? Difference arrays usually mean different "pointer", how did the tests pass?

`toEqual` compares structure only. To compare reference, there is [`toBe`](https://vitest.dev/api/expect.html#tobe). In any case, I think the change is not needed.

Reply:
After updating the libs and running `make test`
we are getting assertion failed 
<img width="1171" alt="image" src="https://github.com/go-gitea/gitea/assets/80308335/252bd0cf-5d72-4d2e-a392-dd5486b19930">

Reply:
Sure, I will share my findings, No worries.

Reply:
Yes, `toEqual` is the right function, the old code is right, it is an upstream bug.

->  0.34.3 doens't work with Uint8Array and TextEncoder https://github.com/vitest-dev/vitest/issues/4043